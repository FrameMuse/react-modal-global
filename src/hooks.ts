/*

MIT License

Copyright (c) 2023 Valery Zinchenko

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

*/

import { useContext, useEffect, useState } from "react"

import { modalContext } from "./context"
import { ModalController } from "./ModalController"
import { ModalWindow } from "./ModalWindow"
import { ModalComponent, ModalState } from "./types"

/**
 * Used inside a modal component to access the modal context (`ModalWindow`).
 *
 * Accepts a generic type that is used to infer the props of the modal component.
 * It has 3 overloads:
 * 1. `useModalWindow<typeof ModalClassComponent>()` - infers the props from the class component type.
 * 2. `useModalWindow<typeof ModalFunctionComponent>()` - infers the props from the function component type.
 * 3. `useModalWindow<{ c: 3 }>()` - you can enter props by yourself too.
 */
export function useModalWindow<T>(): ModalWindow<T extends ModalComponent<infer Props> ? Props : T> {
  const context = useContext(modalContext)
  if (!context) throw new Error(`ModalError: ${useModalWindow.name} must be used within a modal context.`)

  // It's safe to case to `any` here because the context is always set to a `ModalWindow`
  // and the arbitrary type `T` is difened by the user.
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  return context as ModalWindow<any>
}



const DEFAULT_STATE: ModalState = {
  active: false,
  windows: []
}

/**
 * Provides access to modal state and listens to updates.
 */
export function useModalState(controller: ModalController): ModalState {
  const [modalState, setModalState] = useState(DEFAULT_STATE)

  useEffect(() => {
    return controller.observe(setModalState)
  }, [controller])

  return modalState
}
