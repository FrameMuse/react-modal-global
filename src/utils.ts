/**
 *
 * @returns `class1 class2`
 */
export function classMerge(classNames: Array<string | null | undefined>): string {
  const space = " "
  return classNames.filter(Boolean).join(space)
}

/**
 * Join modifiers with origin class
 * @returns `"origin-class origin-class--modifier"`
 */
export function classWithModifiers(originClass: string, ...modifiers: Array<string | number | false | null | undefined>): string {
  modifiers = modifiers.filter(Boolean)
  if (!modifiers.length) return originClass

  const space = " "
  const separator = "--"

  modifiers = modifiers.map(modifier => originClass + separator + modifier)
  return originClass + space + modifiers.join(space)
}
