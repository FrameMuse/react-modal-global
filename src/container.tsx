import React from "react"
import { classWithModifiers } from "./utils"
import { PopupContext } from "./context"
import { PopupWindow } from "./interfaces"
import { PopupPrivate } from "./controller"

export interface PopupContainerProps {
  className?: string
}
export interface PopupContainerState {
  isActive: boolean
  queue: PopupWindow[]
}

export class PopupContainer extends React.Component<PopupContainerProps, PopupContainerState> {
  state: PopupContainerState = {
    isActive: false,
    queue: []
  }

  constructor(props: any) {
    super(props)
    // Set Popup dispatcher
    PopupPrivate.dispatch = this.setState.bind(this)
  }

  render() {
    const { isActive, queue } = this.state
    const lastPopup = queue[queue.length - 1]
    const { component: PopupWindowComponent, params = {}, close } = lastPopup || {}

    const className = this.props.className || "popup"
    return (
      <div className={classWithModifiers(className, isActive && "active")}>
        <div className={className + "__container"} onClick={close}>
          <div className={className + "__inner"} onClick={event => event.stopPropagation()}>
            <PopupContext.Provider value={lastPopup}>
              {PopupWindowComponent && <PopupWindowComponent {...params} />}
            </PopupContext.Provider>
          </div>
        </div>
      </div>
    )
  }
}
