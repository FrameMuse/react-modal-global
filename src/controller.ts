import { Dispatch, SetStateAction } from "react";
import { PopupContainerState } from "./container";
import { PopupComponent, PopupParams, PopupWindow } from "./interfaces"

type AnyIfEmpty<T extends object> = keyof T extends never ? any : T;

export const PopupPrivate: {
  dispatch: Dispatch<SetStateAction<PopupContainerState>>
} = {
  dispatch: () => { }
}

export class Popup {
  public static open
    <P extends object = {}, AC extends Partial<PopupParams> & P = Partial<PopupParams> & P>
    (component: PopupComponent<P>, ...[params]: (AnyIfEmpty<P> extends object ? [AC] : [AC?])): Promise<void> {
    return new Promise<void>(function (resolve) {
      const popupWindow = { component, params, close }
      Popup.addToQueue(popupWindow)
      function close() {
        resolve()
        Popup.removeFromQueue(popupWindow)
      }
    })
  }
  private static addToQueue(popupWindow: PopupWindow<any>) {
    PopupPrivate.dispatch(state => ({
      isActive: true,
      queue: [...state.queue, popupWindow]
    }))
  }
  private static removeFromQueue(popupWindow: PopupWindow<any>) {
    PopupPrivate.dispatch(state => {
      const queue = state.queue.filter(pw => pw !== popupWindow)
      return {
        isActive: queue.length > 0,
        queue
      }
    })
  }
  public static closeAll() {
    PopupPrivate.dispatch(state => {
      state.queue.forEach(popup => popup.close())
      return {
        isActive: false,
        queue: []
      }
    })
  }
}
