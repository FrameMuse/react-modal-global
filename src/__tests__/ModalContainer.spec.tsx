/*

MIT License

Copyright (c) 2023 Valery Zinchenko

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

*/

import "@testing-library/jest-dom"

import { render } from "@testing-library/react"
import { useEffect, useState } from "react"
import { act } from "react-dom/test-utils"

import { ModalContainer } from "../container"
import { ModalController } from "../controller"
import { useModalContext } from "../hooks"
import { classWithModifiers } from "../utils"

let Modal: ModalController = new ModalController()
let container = render(<ModalContainer controller={Modal} />).container
let modalContainerElement = container.querySelector(".modal")

beforeEach(() => {
  Modal = new ModalController()
  container = render(<ModalContainer controller={Modal} />).container
  modalContainerElement = container.querySelector(".modal")
})

describe("ModalContainer", () => {
  it("should render", () => {
    expect(modalContainerElement).not.toBeNull()
    expect(modalContainerElement).toMatchSnapshot()
  })

  it("should have the correct class name by default", () => {
    expect(modalContainerElement).toHaveClass("modal")
    expect(modalContainerElement).not.toHaveClass("modal--open")

    expect(modalContainerElement).toMatchSnapshot()
  })

  it("should have the correct class name when open", () => {
    act(() => {
      Modal.open(PopupExample)
    })

    expect(modalContainerElement).toHaveClass(classWithModifiers("modal", "active"))
    expect(modalContainerElement).toMatchSnapshot()
  })

  it("should have the correct class name when closed (not weak)", () => {
    act(() => {
      Modal.open(PopupExample)
    })

    act(() => {
      modalContainerElement?.querySelector("button")?.click()
    })

    expect(modalContainerElement).not.toHaveClass(classWithModifiers("modal", "active"))
    expect(modalContainerElement).toMatchSnapshot()
  })

  it("should render the correct component", () => {
    act(() => {
      Modal.open(PopupExample)
    })

    expect(modalContainerElement?.querySelector("h1")).toHaveTextContent("Popup Example")
    expect(modalContainerElement).toMatchSnapshot()
  })

  it("should render nullish component", () => {
    act(() => {
      Modal.open(() => null)
    })

    expect(modalContainerElement?.querySelector("h1")).toBeNull()
    expect(modalContainerElement).toMatchSnapshot()
  })

  it("should render the correct component on re-open (not weak)", () => {
    act(() => {
      Modal.open(PopupExample, { random: true })
    })

    const savedElement = modalContainerElement?.cloneNode(true)

    act(() => {
      modalContainerElement?.querySelector("button")?.click()
    })

    act(() => {
      Modal.open(PopupExample, { random: true })
    })

    expect(modalContainerElement).toEqual(savedElement)
  })

  it("should render the correct component on re-open (weak)", () => {
    act(() => {
      Modal.open(PopupExample, { random: true, weak: true })
    })

    const savedElement = modalContainerElement?.cloneNode(true)

    act(() => {
      modalContainerElement?.querySelector("button")?.click()
    })

    act(() => {
      Modal.open(PopupExample, { random: true, weak: true })
    })

    expect(modalContainerElement).not.toEqual(savedElement)
  })

  it("should render the correct component when closed (not weak)", () => {
    act(() => {
      Modal.open(PopupExample)
    })

    act(() => {
      modalContainerElement?.querySelector("button")?.click()
    })

    expect(modalContainerElement?.querySelector("h1")).toHaveTextContent("Popup Example")
    expect(modalContainerElement).toMatchSnapshot()
  })

  it("should render the correct component when closed (weak)", () => {
    act(() => {
      Modal.open(PopupExample, { weak: true })
    })

    act(() => {
      modalContainerElement?.querySelector("button")?.click()
    })

    expect(modalContainerElement?.querySelector("h1")).toBeNull()
    expect(modalContainerElement).toMatchSnapshot()
  })

  it("should render the correct component when closed (weak) and re-opened", () => {
    act(() => {
      Modal.open(PopupExample, { weak: true })
    })

    act(() => {
      modalContainerElement?.querySelector("button")?.click()
    })

    act(() => {
      Modal.open(PopupExample, { weak: true })
    })

    expect(modalContainerElement?.querySelector("h1")).toHaveTextContent("Popup Example")
    expect(modalContainerElement).toMatchSnapshot()
  })

  it("should render the correct component when open multiple and closed one", () => {
    act(() => {
      Modal.open(PopupExample)
      Modal.open(PopupExample, { test: true })
      Modal.open(PopupExample, { test: "1" })
      Modal.open(PopupExample, { test: "2" })
    })

    act(() => {
      modalContainerElement?.querySelector("button")?.click()
    })

    expect(modalContainerElement?.querySelector("#test")).toHaveTextContent("1")
    expect(modalContainerElement).toMatchSnapshot()
  })

  it("should render the correct component when open multiple and closed-open-closed one", () => {
    act(() => {
      const modal1 = Modal.open(PopupExample, { test: "1" })
      modal1.close()
      const modal2 = Modal.open(PopupExample, { test: "2" })
      modal2.close()
    })

    expect(modalContainerElement?.querySelector("h1")).toHaveTextContent("Popup Example")
    expect(modalContainerElement?.querySelector("#test")).toHaveTextContent("2")
    expect(modalContainerElement).toMatchSnapshot()
  })

  it("should render the correct component when open multiple and closed all", () => {
    act(() => {
      Modal.open(PopupExample)
      Modal.open(PopupExample, { test: true })
      Modal.open(PopupExample, { test: "1" })
      Modal.open(PopupExample, { test: "2" })
    })

    act(() => {
      Modal.closeAll()
    })

    expect(modalContainerElement?.querySelector("h1")).toHaveTextContent("Popup Example")
    expect(modalContainerElement?.querySelector("#test")).toHaveTextContent("2")
    expect(modalContainerElement).toMatchSnapshot()
  })

  it("should render the correct component when closed by component 1", () => {
    act(() => {
      Modal.open(PopupExample)
      Modal.open(PopupExample, { test: "7" })
      Modal.open(PopupExample, { test: "1" })
      Modal.open(PopupExample, { test: "2" })

      Modal.closeByComponent(PopupExample)
    })

    expect(modalContainerElement?.querySelector("#test")).toHaveTextContent("2")
    expect(modalContainerElement).toMatchSnapshot()
  })

  it("should render the correct component when closed by component 2", () => {
    act(() => {
      Modal.open(() => null)
      Modal.open(PopupExample)
      Modal.open(PopupExample, { test: "7" })
      Modal.open(PopupExample, { test: "1" })
      Modal.open(PopupExample, { test: "2" })

      Modal.closeByComponent(PopupExample)
    })

    expect(modalContainerElement?.querySelector("h1")).toBeNull()
    expect(modalContainerElement?.querySelector("#test")).toBeNull()
    expect(modalContainerElement).toMatchSnapshot()
  })

  it("should render the correct component when closed by id", () => {
    act(() => {
      Modal.open(PopupExample)
      Modal.open(PopupExample, { test: true })
      Modal.open(PopupExample, { test: "1", id: "1" })
      Modal.open(PopupExample, { test: "2", id: "2" })
      Modal.open(PopupExample, { test: "3", id: "2" })

      Modal.closeById("2")
    })

    expect(modalContainerElement?.querySelector("#test")).toHaveTextContent("1")
    expect(modalContainerElement).toMatchSnapshot()
  })

  it("should work with same repeated modals", () => {
    act(() => {
      Modal.open(PopupExample)
      Modal.open(PopupExample)
    })

    expect(modalContainerElement?.querySelectorAll("h1")).toHaveLength(1)
    expect(modalContainerElement).toMatchSnapshot()
  })

  it("should render the correct component when replaced (open)", () => {
    act(() => {
      Modal.open(PopupExample)
      Modal.replace(PopupExample, { test: true })
    })

    expect(modalContainerElement?.querySelector("h1")).toHaveTextContent("Popup Example")
    expect(modalContainerElement?.querySelector("#test")).not.toBeNull()
    expect(modalContainerElement?.classList).toContain("modal--active")
    expect(modalContainerElement).toMatchSnapshot()
  })

  it("should render the correct component when replaced (closed)", () => {
    act(() => {
      Modal.open(PopupExample)
      const modal = Modal.replace(PopupExample, { test: true })
      modal.close()
    })

    expect(modalContainerElement?.querySelector("h1")).toHaveTextContent("Popup Example")
    expect(modalContainerElement?.querySelector("#test")).not.toBeNull()
    expect(modalContainerElement?.classList).not.toContain("modal--active")
    expect(modalContainerElement).toMatchSnapshot()
  })

  it("should render the correct component when forked", () => {
    act(() => {
      Modal.open(PopupExample)
      Modal.open(PopupExample, { test: true, fork: true })
    })

    expect(container?.querySelectorAll(".modal__container")).toHaveLength(2)
    expect(container?.querySelector("#test")).not.toBeNull()
    expect([...container?.children || []].every(child => child.classList.contains("modal--active"))).toBe(true)
    expect(container).toMatchSnapshot()
  })

  it("should render the correct component when forked closed", () => {
    act(() => {
      Modal.open(PopupExample)
      const modalForked = Modal.open(PopupExample, { test: true, fork: true })
      modalForked.close()
    })

    expect(container?.querySelectorAll(".modal__container")).toHaveLength(1)
    expect(container?.querySelector("#test")).toBeNull()
    expect([...container?.children || []].every(child => child.classList.contains("modal--active"))).toBe(true)
    expect(container).toMatchSnapshot()
  })

  it("should render the correct component when forked (not closable)", () => {
    act(() => {
      Modal.open(PopupExample)
      Modal.open(PopupExample, { test: true, fork: true, closable: false })
    })

    expect(container?.querySelectorAll(".modal__container")).toHaveLength(2)
    expect(container?.querySelector("#test")).not.toBeNull()
    expect([...container?.children || []].every(child => child.classList.contains("modal--active"))).toBe(true)
    expect(container).toMatchSnapshot()
  })
})


function PopupExample(props: { random?: boolean, test?: string | boolean }) {
  const modal = useModalContext()
  const [random, setRandom] = useState(0)
  useEffect(() => setRandom(Date.now()), [props.random])

  return (
    <div>
      {props.random && <span id="random">{random}</span>}
      {props.test && <span id="test">{props.test}</span>}
      <h1>Popup Example</h1>
      <button onClick={modal.close}>Close</button>
    </div>
  )
}
