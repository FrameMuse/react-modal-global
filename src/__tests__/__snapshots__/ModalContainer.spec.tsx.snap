// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalContainer should close on overlay click 1`] = `
<div
  aria-hidden="true"
  aria-modal="true"
  class="modal"
>
  <div
    class="modal__container"
  >
    <div>
      <h1>
        Popup Example
      </h1>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should have the correct class name by default 1`] = `
<div
  aria-hidden="true"
  aria-modal="true"
  class="modal"
/>
`;

exports[`ModalContainer should have the correct class name when closed 1`] = `
<div
  aria-hidden="true"
  aria-modal="true"
  class="modal"
>
  <div
    class="modal__container"
  >
    <div>
      <h1>
        Popup Example
      </h1>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should have the correct class name when open 1`] = `
<div
  aria-hidden="false"
  aria-modal="true"
  class="modal modal--active"
>
  <div
    class="modal__container"
  >
    <div>
      <h1>
        Popup Example
      </h1>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should not close if \`closable\` is \`false\` 1`] = `
<div
  aria-hidden="false"
  aria-modal="true"
  class="modal modal--active"
>
  <div
    class="modal__container"
  >
    <div>
      <h1>
        Popup Example
      </h1>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should render 1`] = `
<div
  aria-hidden="true"
  aria-modal="true"
  class="modal"
/>
`;

exports[`ModalContainer should render nullish component 1`] = `
<div
  aria-hidden="false"
  aria-modal="true"
  class="modal modal--active"
>
  <div
    class="modal__container"
  />
</div>
`;

exports[`ModalContainer should render the correct component 1`] = `
<div
  aria-hidden="false"
  aria-modal="true"
  class="modal modal--active"
>
  <div
    class="modal__container"
  >
    <div>
      <h1>
        Popup Example
      </h1>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should render the correct component on re-open 1`] = `
<div
  aria-hidden="false"
  aria-modal="true"
  class="modal modal--active"
>
  <div
    class="modal__container"
  >
    <div>
      <span
        id="test"
      >
        1
      </span>
      <h1>
        Popup Example
      </h1>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should render the correct component on re-open 2`] = `
<div
  aria-hidden="false"
  aria-modal="true"
  class="modal modal--active"
>
  <div
    class="modal__container"
  >
    <div>
      <span
        id="test"
      >
        1
      </span>
      <h1>
        Popup Example
      </h1>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should render the correct component when closed 1`] = `
<div
  aria-hidden="true"
  aria-modal="true"
  class="modal"
>
  <div
    class="modal__container"
  >
    <div>
      <h1>
        Popup Example
      </h1>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should render the correct component when closed by component 1 1`] = `
<div
  aria-hidden="true"
  aria-modal="true"
  class="modal"
>
  <div
    class="modal__container"
  >
    <div>
      <span
        id="test"
      >
        2
      </span>
      <h1>
        Popup Example
      </h1>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should render the correct component when closed by component 2 1`] = `
<div
  aria-hidden="false"
  aria-modal="true"
  class="modal modal--active"
>
  <div
    class="modal__container"
  />
</div>
`;

exports[`ModalContainer should render the correct component when closed by id 1`] = `
<div
  aria-hidden="false"
  aria-modal="true"
  class="modal modal--active"
>
  <div
    class="modal__container"
  >
    <div>
      <span
        id="test"
      >
        1
      </span>
      <h1>
        Popup Example
      </h1>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should render the correct component when forked (not closable) 1`] = `
<div>
  <div
    aria-hidden="false"
    aria-modal="true"
    class="modal modal--active"
  >
    <div
      class="modal__container"
    >
      <div>
        <h1>
          Popup Example
        </h1>
        <button>
          Close
        </button>
      </div>
    </div>
    <div
      class="modal__container"
    >
      <div>
        <span
          id="test"
        />
        <h1>
          Popup Example
        </h1>
        <button>
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should render the correct component when forked 1`] = `
<div>
  <div
    aria-hidden="false"
    aria-modal="true"
    class="modal modal--active"
  >
    <div
      class="modal__container"
    >
      <div>
        <h1>
          Popup Example
        </h1>
        <button>
          Close
        </button>
      </div>
    </div>
    <div
      class="modal__container"
    >
      <div>
        <span
          id="test"
        />
        <h1>
          Popup Example
        </h1>
        <button>
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should render the correct component when forked closed 1`] = `
<div>
  <div
    aria-hidden="false"
    aria-modal="true"
    class="modal modal--active"
  >
    <div
      class="modal__container"
    >
      <div>
        <h1>
          Popup Example
        </h1>
        <button>
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should render the correct component when open multiple and closed all 1`] = `
<div
  aria-hidden="true"
  aria-modal="true"
  class="modal"
>
  <div
    class="modal__container"
  >
    <div>
      <span
        id="test"
      >
        2
      </span>
      <h1>
        Popup Example
      </h1>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should render the correct component when open multiple and closed one 1`] = `
<div
  aria-hidden="false"
  aria-modal="true"
  class="modal modal--active"
>
  <div
    class="modal__container"
  >
    <div>
      <span
        id="test"
      >
        1
      </span>
      <h1>
        Popup Example
      </h1>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should render the correct component when open multiple and closed-open-closed one 1`] = `
<div
  aria-hidden="true"
  aria-modal="true"
  class="modal"
>
  <div
    class="modal__container"
  >
    <div>
      <span
        id="test"
      >
        2
      </span>
      <h1>
        Popup Example
      </h1>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should render the correct component when open the same window and closed 1`] = `
<div
  aria-hidden="true"
  aria-modal="true"
  class="modal"
>
  <div
    class="modal__container"
  >
    <div>
      <span
        id="test"
      >
        2
      </span>
      <h1>
        Popup Example
      </h1>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should render the correct component when replaced (closed) 1`] = `
<div
  aria-hidden="true"
  aria-modal="true"
  class="modal"
>
  <div
    class="modal__container"
  >
    <div>
      <span
        id="test"
      />
      <h1>
        Popup Example
      </h1>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should render the correct component when replaced (open) 1`] = `
<div
  aria-hidden="false"
  aria-modal="true"
  class="modal modal--active"
>
  <div
    class="modal__container"
  >
    <div>
      <span
        id="test"
      />
      <h1>
        Popup Example
      </h1>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalContainer should work with same repeated modals 1`] = `
<div
  aria-hidden="false"
  aria-modal="true"
  class="modal modal--active"
>
  <div
    class="modal__container"
  >
    <div>
      <h1>
        Popup Example
      </h1>
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;
